search-app/src/cmr/search/middleware/shapefile.clj

```diff
@@ -0,0 +1,61 @@
+(ns cmr.search.middleware.shapefile
+  "Contains parameter converters for shapefile parameter"
+  (:require 
+    [clojure.java.io :as io]
+    [clojure.string :as str]
+    [ring.middleware.multipart-params :refer [wrap-multipart-params]]
+    [cmr.common.config :as cfg :refer [defconfig]]
+    [cmr.common.api.errors :as api-errors]
+    [cmr.common.log :refer [debug]]
+    [cmr.common.mime-types :as mt]
+    [cmr.common-app.services.search.group-query-conditions :as gc]
+    [cmr.common-app.services.search.params :as p]
+    [cmr.common.services.errors :as errors]
+    [cmr.search.models.query :as qm]
+    [cmr.search.services.parameters.converters.geojson :as geojson]
+    [cmr.search.services.parameters.converters.geometry :as geo]
+    [cmr.common.util :as util])
+  (:import
+   (java.io BufferedInputStream File FileReader FileOutputStream FileInputStream)
+   (java.nio.file Files)
+   (java.nio.file.attribute FileAttribute)
+   (java.net URL)
+   (java.util HashMap)
+   (java.util.zip ZipFile ZipInputStream)
+   (org.apache.commons.io FilenameUtils)
+   (org.geotools.data DataStoreFinder FileDataStoreFinder Query)
+   (org.geotools.data.simple SimpleFeatureSource)
+   (org.geotools.data.geojson GeoJSONDataStore)
+   (org.geotools.geometry.jts JTS)
+   (org.geotools.kml.v22 KMLConfiguration KML)
+   (org.geotools.referencing CRS)
+   (org.geotools.util URLs)
+   (org.geotools.xsd Parser StreamingParser PullParser)
+   (org.opengis.feature.simple SimpleFeature)
+   (org.opengis.feature.type Name)))
+
+(defconfig max-shapefile-size
+  "The maximum size in bytes a shapefile can be"
+  {:default 1000000 :type Long})
+
+(defn- progress
+  "Progress function for `wrap-multipart-params`. This function simply throws an error if
+  the uploaded file exceeds a given limit."
+  [_reqest _bytes-read content-length _item-count]
```

Comment:
reqest -> request. _Maybe_ useful to also log the `content-length` a user attempted to upload on failure.

Reply:
done